go/cmd/dolt/commands/commit.go

```diff
@@ -103,176 +113,150 @@ func performCommit(ctx context.Context, commandStr string, args []string, dEnv *
-	}
-
-	roots, err := dEnv.Roots(ctx)
+	queryist, sqlCtx, closeFunc, err := cliCtx.QueryEngine(ctx)
 	if err != nil {
-		return HandleVErrAndExitCode(errhand.BuildDError("Couldn't get working root").AddCause(err).Build(), usage), false
+		cli.Println(err.Error())
+		return 1, false
 	}
-
-	if upperCaseAllFlag {
-		roots, err = actions.StageAllTables(ctx, roots, true)
-		if err != nil {
-			return handleCommitErr(ctx, dEnv, err, help), false
-		}
-	} else if allFlag {
-		roots, err = actions.StageModifiedAndDeletedTables(ctx, roots)
-		if err != nil {
-			return handleCommitErr(ctx, dEnv, err, help), false
-		}
-	}
-
-	headCommit, _ := dEnv.HeadCommit(ctx)
-	headHash, _ := headCommit.HashOf()
-
-	var name, email string
-	// Check if the author flag is provided otherwise get the name and email stored in configs
-	if authorStr, ok := apr.GetValue(cli.AuthorParam); ok {
-		name, email, err = cli.ParseAuthor(authorStr)
-	} else {
-		// This command creates a commit, so we need user identity
-		if !cli.CheckUserNameAndEmail(dEnv) {
-			return 1, false
-		}
-		name, email, err = env.GetNameAndEmail(dEnv.Config)
+	if closeFunc != nil {
+		defer closeFunc()
 	}
 
+	err = branch_control.CheckAccess(sqlCtx, branch_control.Permissions_Write)
 	if err != nil {
-		return handleCommitErr(ctx, dEnv, err, usage), false
+		cli.Println(err.Error())
+		return 1, false
 	}
 
 	msg, msgOk := apr.GetValue(cli.MessageArg)
 	if !msgOk {
 		amendStr := ""
 		if apr.Contains(cli.AmendFlag) {
-			commitMeta, cmErr := headCommit.GetCommitMeta(ctx)
-			if cmErr != nil {
-				return handleCommitErr(ctx, dEnv, cmErr, usage), false
+			_, rowIter, err := queryist.Query(sqlCtx, "select message from dolt_log() limit 1")
+			if err != nil {
+				cli.Println(err.Error())
+				return 1, false
 			}
-			amendStr = commitMeta.Description
+			row, err := rowIter.Next(sqlCtx)
+			if err != nil {
+				cli.Println(err.Error())
+				return 1, false
+			}
+			amendStr = row[0].(string)
 		}
-		msg, err = getCommitMessageFromEditor(ctx, dEnv, "", amendStr, false)
+		msg, err = getCommitMessageFromEditor(ctx, sqlCtx, queryist, "", amendStr, false, cliCtx)
 		if err != nil {
-			return handleCommitErr(ctx, dEnv, err, usage), false
+			return handleCommitErr(sqlCtx, queryist, err, usage), false
 		}
 	}
 
-	t := datas.CommitNowFunc()
-	if commitTimeStr, ok := apr.GetValue(cli.DateParam); ok {
-		var err error
-		t, err = cli.ParseDate(commitTimeStr)
+	// process query through prepared statement to prevent sql injection
+	query, params := callDoltCommitStoredProc(msg, apr)
+	q, err := dbr.InterpolateForDialect(query, params, dialect.MySQL)
+	if err != nil {
+		cli.Println(err.Error())
+		return 1, false
+	}
 
-		if err != nil {
-			return HandleVErrAndExitCode(errhand.BuildDError("error: invalid date").AddCause(err).Build(), usage), false
-		}
+	_, _, err = queryist.Query(sqlCtx, q)
```

Comment:
Rather than drop the results and follow up with a log call, let's grab the hash here in the non-empty case. I _think_ you can skip the log call most of the time. I'm not sure, I'll leave that investigation up to you.

Reply:
I'm not sure we need the hash here, once `log` is migrated, we can switch back to a call of `dolt log -n-1` which wouldn't need the hash anyways. With the log call we can print the message as well which some of our bats tests check for.